### ISSUE TITLE
ext.todo: launches exception KeyError: 'refid'

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#902]
Here's the traceback (while writing the page that contains the //.. todolist:://):

```
# Sphinx version: 1.1.2
# Python version: 2.7.2+
# Docutils version: 0.7 release
# Jinja2 version: 2.6
Traceback (most recent call last):
  File "/usr/lib/pymodules/python2.7/sphinx/cmdline.py", line 189, in main
    app.build(force_all, filenames)
  File "/usr/lib/pymodules/python2.7/sphinx/application.py", line 204, in build
    self.builder.build_update()
  File "/usr/lib/pymodules/python2.7/sphinx/builders/__init__.py", line 196, in build_update
    'out of date' &#37; len(to_build))
  File "/usr/lib/pymodules/python2.7/sphinx/builders/__init__.py", line 252, in build
    self.write(docnames, list(updated_docnames), method)
  File "/usr/lib/pymodules/python2.7/sphinx/builders/__init__.py", line 291, in write
    doctree = self.env.get_and_resolve_doctree(docname, self)
  File "/usr/lib/pymodules/python2.7/sphinx/environment.py", line 1228, in get_and_resolve_doctree
    self.resolve_references(doctree, docname, builder)
  File "/usr/lib/pymodules/python2.7/sphinx/environment.py", line 1475, in resolve_references
    builder.app.emit('doctree-resolved', doctree, fromdocname)
  File "/usr/lib/pymodules/python2.7/sphinx/application.py", line 314, in emit
    results.append(callback(self, *args))
  File "/usr/lib/pymodules/python2.7/sphinx/ext/todo.py", line 123, in process_todo_nodes
    newnode['refuri'] += '#' + todo_info['target']['refid']
  File "/usr/lib/pymodules/python2.7/docutils/nodes.py", line 534, in __getitem__
    return self.attributes[key]
KeyError: 'refid'
```

Tried also with 1.1.3 and the same exception appears.

Thanks

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/902
- Originally reported by: Anonymous
- Originally created at: 2012-03-20T17:03:30.104

**Comment by shimizukawa** [comment]
_From Lluís Vilanova on 2013-01-16 18:52:16+00:00_

The same has also been (and is) happening to me for some time.

**Comment by shimizukawa** [comment]
_From Philippe Martin on 2013-10-09 15:10:40+00:00_

Hello,

I am using sphinx 1.2b1 and have the same issue.

Regards,

Philippe

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-11-16 09:27:16+00:00_

Please let me know the small documentation sample to reproduce this issue. Thanks.

**Comment by shimizukawa** [comment]
_From Ryo OKUBO on 2013-11-19 16:01:31+00:00_

Hello,

I also encountered a similar problem when build documents with po files that contains auto-numbered footnotes.

Documentation samples:
https://gist.github.com/syucream/7547504/raw/e4c9ff29119f084eda70169fe2034ad4e03b1f63/index.rst
https://gist.github.com/syucream/7547481/raw/58da7c050419c4d7bf7235061017122f0440c1f0/index.po

Command logs:
https://gist.github.com/syucream/7547023/raw/e3b7722a988b58bf1fc4303f5984e9c727ce9f5a/sphinx-build-error-log

Thanks.

**Comment by shimizukawa** [comment]
_From Ryo OKUBO on 2013-11-20 12:33:45+00:00_

Sorry, my problem has already been fixed in #955:
